tests/testsuite/metadata.rs

```diff
@@ -627,6 +627,497 @@ fn cargo_metadata_with_deps_and_version() {
+          "registry+https://github.com/rust-lang/crates.io-index#bar@0.0.1"
+        ],
+        "deps": [
+          {
+            "dep_kinds": [
+              {
+                "kind": null,
+                "target": null
+              }
+            ],
+            "name": "bar",
+            "pkg": "registry+https://github.com/rust-lang/crates.io-index#bar@0.0.1"
+          }
+        ],
+        "features": [],
+        "id": "path+[ROOTURL]/foo#0.5.0"
+      }
+    ],
+    "root": "path+[ROOTURL]/foo#0.5.0"
+  },
+  "target_directory": "[ROOT]/foo/target",
+  "version": 1,
+  "workspace_default_members": [
+    "path+[ROOTURL]/foo#0.5.0"
+  ],
+  "workspace_members": [
+    "path+[ROOTURL]/foo#0.5.0"
+  ],
+  "workspace_root": "[ROOT]/foo"
+}
+"#]]
+            .is_json(),
+        )
+        .run();
+}
+
+#[cargo_test]
+fn cargo_metadata_public_private_dependencies_enabled() {
+    let p = project()
+        .file("src/foo.rs", "")
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.5.0"
+                authors = []
+                license = "MIT"
+                description = "foo"
+
+                [[bin]]
+                name = "foo"
+
+                [dependencies]
+                bar = { version = "*", public = false }
+                foobar = { version = "*", public = true }
+                baz = "*"
+            "#,
+        )
+        .build();
+    Package::new("bar", "0.0.1").publish();
+    Package::new("foobar", "0.0.2").publish();
+    Package::new("baz", "0.0.3").publish();
+
+    p.cargo("metadata -q --format-version 1 -Zpublic-dependency")
+        .masquerade_as_nightly_cargo(&["public-dependency"])
+        .with_stdout_data(
+            str![[r#"
+{
+  "metadata": null,
+  "packages": [
+    {
+      "authors": [],
+      "categories": [],
+      "default_run": null,
+      "dependencies": [],
+      "description": null,
+      "documentation": null,
+      "edition": "2015",
+      "features": {},
+      "homepage": null,
+      "id": "registry+https://github.com/rust-lang/crates.io-index#bar@0.0.1",
+      "keywords": [],
+      "license": null,
+      "license_file": null,
+      "links": null,
+      "manifest_path": "[ROOT]/home/.cargo/registry/src/-[HASH]/bar-0.0.1/Cargo.toml",
+      "metadata": null,
+      "name": "bar",
+      "publish": null,
+      "readme": null,
+      "repository": null,
+      "rust_version": null,
+      "source": "registry+https://github.com/rust-lang/crates.io-index",
+      "targets": [
+        {
+          "crate_types": [
+            "lib"
+          ],
+          "doc": true,
+          "doctest": true,
+          "edition": "2015",
+          "kind": [
+            "lib"
+          ],
+          "name": "bar",
+          "src_path": "[ROOT]/home/.cargo/registry/src/-[HASH]/bar-0.0.1/src/lib.rs",
+          "test": true
+        }
+      ],
+      "version": "0.0.1"
```

Comment:
Could you redact the fields that are not relevant for these tests?  I'd recommend keeping all keys in a `dependencies` table but most package metadata keys and the `resolve` graph could probably go

#### Examples

Redact a value
```json
  "workspace_members": "{...}",
```

Redact all other keys in a table
```json
"...": "{...}"
```

Reply:
Oh nice!

Addressed with a force push